numba/targets/arraymath.py

```diff
@@ -2782,6 +2782,52 @@ def np_imag_impl(a):
 #----------------------------------------------------------------------------
 # Misc functions
 
+@overload(np.delete)
+def np_delete(arr, obj):
+    # Implementation based on numpy
+    # https://github.com/numpy/numpy/blob/af66e487a57bfd4850f4306e3b85d1dac3c70412/numpy/lib/function_base.py#L4065-L4267
+
+    if not isinstance(arr, (types.Array, types.Sequence)):
+        raise TypingError("arr must be either an Array or a Sequence")
+
+    def np_delete_impl(arr, obj):
+        arr = np.ravel(np.asarray(arr))
+        N = arr.size
+
+        keep = np.ones(N, dtype=np.bool_)
+        obj = handler(obj)
+        keep[obj] = False
+        return arr[keep]
+
+    def np_delete_scalar_impl(arr, obj):
+        arr = np.ravel(np.asarray(arr))
+        N = arr.size
+        pos = obj
+
+        if (pos < -N or pos >= N):
+            raise IndexError('obj must be less than the len(arr)')
+            # NumPy raises IndexError: index 'i' is out of
+            # bounds for axis 'x' with size 'n'
+
+        if (pos < 0):
+            pos += N
+
+        return np.concatenate((arr[:pos], arr[pos+1:]))
+
+    if isinstance(obj, (types.Array, types.Sequence, types.SliceType)):
+        if isinstance(obj, (types.SliceType)):
+            handler = register_jitable(lambda x : x)
```

Comment:
The issue with having these functions defined here is that every single variation over types that cause compilation of a `np.delete` impl will get its own version of one of these functions.

Reply:
https://github.com/numba/numba/commit/0e52c0d13c81ede81c77326fbd4a5ab6dfbf0bea is one way to avoid this.